src/App.js

```diff
@@ -984,18 +987,23 @@ export class App extends React.Component<AppProps, AppState> {
             const serializedProgram = this.programSerializer.serialize(this.state.programSequence.getProgram());
             const serializedCharacterState = this.characterStateSerializer.serialize(this.state.characterState);
             const serializedAllowedActions = this.allowedActionsSerializer.serialize(this.state.allowedActions);
-            window.history.pushState(
-                {
-                    p: serializedProgram,
-                    c: serializedCharacterState,
-                    t: this.state.settings.theme,
-                    a: serializedAllowedActions,
-                    w: this.state.settings.world
-                },
-                '',
-                Utils.generateEncodedProgramURL(this.version, this.state.settings.theme, this.state.settings.world, serializedProgram, serializedCharacterState, serializedAllowedActions),
-                '',
-            );
+            if (this.pushStateTimer == null) {
```

Comment:
This addresses the problem, but in thinking about it more I have a concern that we'd end up not persisting the last change in the series, which is the most critical to preserve.

I'd suggest inverting the logic, if `this.pushStateTimer` is not `null` (i.e. the last update hasn't completed), use `clearTimeout(this.pushStateTimer)` to stop the previous timer, and then always set `this.pushStateTimer` as you're doing here.

That way the last change (the final state) has a better chance of winning if there's a "race".

Another strategy would be to make it queue-based, i.e. push all the state changes to a stack and pop one off every 300ms until there are none left.  Probably that's a bit much for this go.

Reply:
> Another strategy would be to make it queue-based, i.e. push all the state changes to a stack and pop one off every 300ms until there are none left. Probably that's a bit much for this go.

I thought more about this, and @simonbates also noticed that the queue strategy would push the out the time at which all events are persisted and make it more likely that we'd lose things.

Reply:
Thanks @the-t-in-rtf and @simon-idrc, for the feedback, I will rework on the logic to ensure last state gets stored.